services/api-server/tests/unit/api_functions/test_api_routers_functions.py

```diff
@@ -0,0 +1,496 @@
+# pylint: disable=unused-argument
+# pylint: disable=redefined-outer-name
+
+from uuid import uuid4
+
+import pytest
+from httpx import AsyncClient
+from models_library.api_schemas_webserver.functions_wb_schema import (
+    FunctionIDNotFoundError,
+    JSONFunctionInputSchema,
+    JSONFunctionOutputSchema,
+)
+from pytest_mock import MockerFixture
+from simcore_service_api_server._meta import API_VTAG
+
+
+async def test_register_function(
+    client: AsyncClient,
+    mock_function_register: MockerFixture,
+) -> None:
+    sample_function = {
+        "title": "test_function",
+        "function_class": "project",
+        "project_id": str(uuid4()),
+        "description": "A test function",
+        "input_schema": JSONFunctionInputSchema().model_dump(),
+        "output_schema": JSONFunctionOutputSchema().model_dump(),
+        "default_inputs": None,
+    }
+
+    response = await client.post(f"{API_VTAG}/functions", json=sample_function)
+    assert response.status_code == 200
+    data = response.json()
+    assert data["uid"] is not None
+    assert data["function_class"] == sample_function["function_class"]
+    assert data["project_id"] == sample_function["project_id"]
+    assert data["input_schema"] == sample_function["input_schema"]
+    assert data["output_schema"] == sample_function["output_schema"]
+    assert data["title"] == sample_function["title"]
+    assert data["description"] == sample_function["description"]
+
+
+async def test_register_function_invalid(
+    client: AsyncClient, mock_function_register: MockerFixture
+) -> None:
+    invalid_function = {
+        "title": "test_function",
+        "function_class": "invalid_class",  # Invalid class
+        "project_id": str(uuid4()),
+    }
+    response = await client.post(f"{API_VTAG}/functions", json=invalid_function)
+    assert response.status_code == 422  # Unprocessable Entity
+    assert (
+        "Input tag 'invalid_class' found using 'function_class' does not"
+        in response.json()["errors"][0]["msg"]
+    )
+
+
+async def test_get_function(
+    client: AsyncClient, mock_function_register: MockerFixture
+) -> None:
+    project_id = str(uuid4())
```

Comment:
We have faker plugins that I hihgly recommend you to use. eg. in  api-server we include `faker_projects_data`, which already defines a `project_id` fixture

![image](https://github.com/user-attachments/assets/ff29836f-9adb-4bc3-9489-55f0a1a4d98e)


https://github.com/itisfoundation/osparc-simcore/blob/5732a12e07e63d5ce55010ede9b9ab543bb9b278/packages/pytest-simcore/src/pytest_simcore/faker_projects_data.py


See other faker_* fixtures in place.

Reply:
Ok, am using project_id now conftest fixtures.